File: examples/ruby/list-configs.rb
#!/usr/bin/ruby

require "dbus"

system_bus = DBus::SystemBus.instance

service = system_bus.service("org.opensuse.Snapper")

dbus_object = service.object("/org/opensuse/Snapper")

dbus_object.introspect
dbus_object.default_iface = "org.opensuse.Snapper"

configs = dbus_object.send("ListConfigs")[0]

configs.each do |config|
  print config[0], " ", config[1], "\n"
end

File: examples/python/list-configs.py
#!/usr/bin/python3

import dbus

bus = dbus.SystemBus()

snapper = dbus.Interface(bus.get_object('org.opensuse.Snapper', '/org/opensuse/Snapper'),
                         dbus_interface='org.opensuse.Snapper')


configs = snapper.ListConfigs()

for config in configs:
    print(config[0], config[1])
